python/pfs/drp/stella/createRefSpec.py

```diff
@@ -0,0 +1,118 @@
+from lsst.pipe.base import Task, Struct, TaskRunner, ArgumentParser, CmdLineTask
+import lsst.daf.persistence.butler as lsstButler
+import numpy as np
+from astropy.io import fits as pyfits
+import matplotlib.pyplot as plt
+
+class CreateRefSpecConfig(Config):
+    """Configuration for creating the reference spectrum"""
+    lineList = Field( doc = "Reference line list including path", dtype = str, default="/Users/azuri/stella-git/obs_pfs/pfs/lineLists/CdHgKrNeXe_red.fits")
+    output = Field( doc = "Name of output file", dtype=str, default="/Users/azuri/stella-git/obs_pfs/pfs/lineLists/refCdHgKrNeXe_red_temp.fits")
+
+class CreateRefSpecTaskRunner(TaskRunner):
+    """Get parsed values into the CreateRefSpecTask.run"""
+    @staticmethod
+    def getTargetList(parsedCmd, **kwargs):
+        return [dict(expRefList=parsedCmd.id.refList, butler=parsedCmd.butler, lineList=parsedCmd.lineList)]
+
+    def __call__(self, args):
+        task = self.TaskClass(config=self.config, log=self.log)
+        if self.doRaise:
+            self.log.info('CreateRefSpecTask.__call__: args = %s' % args)
+            result = task.run(**args)
+        else:
+            try:
+                result = task.run(**args)
+            except Exception, e:
+                task.log.fatal("Failed: %s" % e)
+
+        if self.doReturnResults:
+            return Struct(
+                args = args,
+                metadata = task.metadata,
+                result = result,
+            )
+
+class CreateRefSpecTask(CmdLineTask):
+    """Task to reduce Arc images"""
+    ConfigClass = CreateRefSpecConfig
+    RunnerClass = CreateRefSpecTaskRunner
+    _DefaultName = "createRefSpecTask"
+
+    def __init__(self, *args, **kwargs):
+        super(CreateRefSpecTask, self).__init__(*args, **kwargs)
+
+    @classmethod
+    def _makeArgumentParser(cls, *args, **kwargs):
+        parser = ArgumentParser(name=cls._DefaultName)
+        parser.add_id_argument("--id", datasetType="postISRCCD",
+                               help="input identifiers, e.g., --id visit=123 ccd=4")
+        parser.add_argument("--lineList", help='directory and name of line list')
+        return parser
+
+    def run(self, expRefList, butler, lineList=None, immediate=True):
+        print 'expRefList = ',expRefList
+        print 'expRefList[0] = ',expRefList[0]
+        arcRef = expRefList[0]
+        try:
+            arcExp = arcRef.get("postISRCCD", immediate=True)
+        except Exception, e:
+            raise RuntimeError("Unable to retrieve %s: %s" % (arcRef.dataId, e))
+
+	try:
+            flatExposure = arcRef.get('flat', immediate=immediate)
+        except Exception, e:
+            raise RuntimeError("Unable to retrieve flat for %s: %s" % (arcRef.dataId, e))
+
+        """ find and trace fiber traces """
+        print 'tracing flat fiber traces'
+        myFindTask = fataTask.FindAndTraceAperturesTask()
```

Comment:
As a total neophyte here, I find the naming a bit confusing. Find and trace fibre traces? Or *flat* fibre traces? Or apertures? Are they all the same? (I assume the answer is yes, but could we at least use the same terminology on the three consecutive lines?)